backend/utils/chat_notification.py

```diff
@@ -0,0 +1,218 @@
+        for DM room.
+        Args:
+            (i) org_id
+            (ii) room_id
+            (iii) sender_id
+            (iv) message
+        Raise:
+            HTTP_422- when novu failed to create DM notification
+        """
+        payload = {}
+        room = get_room_members(org_id,room_id)
+        if not room:
+            return ValueError(
+                "Room not found"
+            )
+         #populate the payload dictionary    
+        payload['sender'] = sender_id
+        payload['message'] = message
+        # create novu subscription for room members in the channel 
+        # if none exist
+        for member_id, values in room.items():
+            if member_id == sender_id:
+                del room[member_id]
+            dm_trigger_create = await event.trigger(
+                '<REPLACE_WITH_EVENT_NAME_FROM_ADMIN_PANEL>',
+                {
+                    "to": {
+                    "subscriberId": member_id
+                    },
+                    "payload": payload
+                    })
+            if dm_trigger_create['statusCode'] !=201:
+                raise HTTPException(
+                status_code=422, 
+                detail="Novu couldn't create DM notification trigger"
+                )
+        return dm_trigger_create
+        
+
+
+    async def channel_messages_trigger(self, message_obj):
+        """
+        A function that triggers a Novu notification instance 
+        for every user in a room exclding the sender.
+        Args:
+            (i) message object->Dict(The message instance in the room)
+        Raise:
+            HTTP_422- when novu failed to create DM notification
+        """
+        payload = {}
+        room_member_list = []
+        org_id = message_obj.get("org_id", '')
+        room_id = message_obj.get("room_id", '')
+        if not org_id or room_id or message_obj:
+            raise HTTPException(
+                status_code=400, 
+                detail="Invalid data input"
+                )
+        try:
+            message = message_obj['data']['blocks'][0]['text']
+            sender_id = message_obj.get("sender_id")
+        except:
+            pass
+        room = get_room(org_id, room_id)
+        if not room:
+            return ValueError("room with ID not found")
+        # create a notfication for the DM user 
+        if room['roomt_type'] == 'DM':
+            dm_create = self.dm_message_trigger(org_id,room_id,sender_id)
+            if dm_create['statusCode'] !=201:
+                raise HTTPException(
+                    status_code=422, 
+                    detail="Failed to process Novu instance"
+                    )
+            if dm_create["acknowledged"] != "true":
+                raise HTTPException(
+                status_code=422, 
+                detail="failed to create a DM Novu instance"
+                )
+            return "dm notification trigger successful"
```

Comment:
the function says `channel_messages_trigger`, why are we handling dm notifications here?

Reply:
when a message object is created, the function differentiates the messages and then call either the method responsible for DM notification(trigger) or channel. I've changed the name @mikengr